### ISSUE TITLE
DatePicker: No Selection

### ISSUE THREAD

**Issue by BarryCarlyon** [Shopify/polaris#351]
Before the update to v2.1.2 I was passing `false` to the `selected` prop of `DatePicker`

This no longer works, results in various levels of errors in `dates.js` function `isSameDay` (javascript-utiltities module)

What is the correct way to instantiate a DatePicker with no date or date range preselected? Or is this a new DatePicker bug after fixing the weird focus bug?

**Comment by chloerice** [comment]
Hi @BarryCarlyon, thanks for raising this. The `selected` prop is not required, so you can pass in `undefined` if you'd like the date picker to start with the date unset. That said, it only accepts a Date or a Range (and object with a 'start' Date and an 'end' Date), so I'm unsure of why there wasn't an error previous to the bug fix released with v2.1.2 because I would've expected one if the value given was a boolean.


<details>
<summary>Click to view collapsed example code using undefined to initialize the date picker with an unset date</summary>

```jsx
import * as React from 'react';
import {Page, AppProvider, DatePicker, Range, Card} from '@shopify/polaris';

export default class Playground extends React.Component {
  state = {
    range: undefined,
    visibleMonth: new Date().getMonth(),
    visibleYear: new Date().getFullYear(),
  };

  render() {
    const {visibleMonth, visibleYear, range} = this.state;
    const now = new Date();

    return (
      <AppProvider>
        <Page title="Playground">
          <Card
            title="Date picker, allows range"
            primaryFooterAction={{
              content: 'Reset to today',
              onAction: this.resetDate('today'),
            }}
            secondaryFooterAction={{
              content: 'Reset',
              onAction: this.resetDate(),
            }}
          >
            <Card.Section>
              <DatePicker
                month={visibleMonth}
                year={visibleYear}
                onChange={this.handleDatePickerChange}
                disableDatesAfter={now}
                onMonthChange={this.handleMonthChange}
                selected={range}
                allowRange
              />
            </Card.Section>
          </Card>
        </Page>
      </AppProvider>
    );
  }

  resetDate = (today) => () => {
    if (!today) {
      return this.setState({
        range: undefined,
        visibleMonth: new Date().getMonth(),
        visibleYear: new Date().getFullYear(),
      });
    }

    this.setState({
      range: {start: new Date(), end: new Date()},
      visibleMonth: new Date().getMonth(),
      visibleYear: new Date().getFullYear(),
    });
  };

  handleDatePickerChange = ({start, end}) => {
    this.setState({range: {start, end}});
  };

  handleMonthChange = (visibleMonth, visibleYear) => {
    this.setState({
      visibleMonth,
      visibleYear,
    });
  };
}
```
</details>

**Comment by BarryCarlyon** [comment]
The only other thing I changed that I can think of is that I updated node today, to v10.4.0 *shrugs*
